src/lib/operators/maintenance/create_table.cpp

```diff
@@ -41,18 +43,19 @@ const std::string CreateTable::description(DescriptionMode description_mode) con
 }
 
 std::shared_ptr<const Table> CreateTable::_on_execute() {
-  // TODO(anybody) chunk size and mvcc not yet specifiable
-  const auto table = std::make_shared<Table>(column_definitions, TableType::Data, Chunk::DEFAULT_SIZE, UseMvcc::Yes);
-
-  StorageManager::get().add_table(table_name, table);
-
+  // If IF NOT EXISTS is not set and the table already exists, StorageManager throws an exception
```

Comment:
What happens if I do it in the console? Optimally, typical user input errors shouldn't kill the entire system.

Reply:
The system is killed in fact if a user does that. This happens because `CreateTable` is strangely enough an `AbstractReadOnlyOperator`. It could become an `AbstractReadWriteOperator`, which wraps the execution in a try-catch block and comes with a state, but then it would also need a transaction context. What do you think?

Reply:
We agreed on a fail-load policy for Hyrise, meaning that every failing `Assert` winds down the system. We weakened this with `AssertInput`, but this is to be used sparingly and probably only in the `SQLTranslator`. Definitely not in Operators. 

The Console is the only component of Hyrise that we want to survive user input errors. The Console, however, is not important enough to warrant weakening the policy that _we do not handle exceptions in operators (in fact, we almost never handle them)_. If the console wants to survive invalid input, it needs to validate the input itself. In this case this means that the console should check generated LQPs/PQPs for Create* operators and validate whether they can be executed.

Another point: The `AbstractReadOnlyOperator` and `AbstractReadWriteOperator` classes are easy to misunderstand in their naming. The "Read" and the "Write" refer **only** to row/MVCC data. 

(Regarding exceptions: I could write a looong rant here about how using them almost always leads to badly tested code, badly readable code, unexpected behaviour and swallowed errors. I could dig up an old version of the `SQLPipeline/Statement` - which used to handle exceptions - to prove this point. It was just terrible.)

Reply:
Thanks for the explanation, @mrzzzrm! 
So, if we want the console to check this, we would need it to do so at least for all 7 maintenance operators (CreatePreparedPlan, CreateTable, CreateView, DropTable, DropView, ShowColumns, ShowTables... There is no DropPreparedPlan-Operator implemented yet, is it?) if not more. 

Should we open a new issue for this?

Reply:
> So, if we want the console to check this, we would need it to do so at least for all 7 maintenance operators.

Yes. Not that hard, though:

```c++
bool verify_plan(lqp) {
  visit_lqp(lqp, [&](const auto& node) {
    switch (node.type) {
      case LQPNodeType::CreateTable:
        return // check whether table already exists
    }
  });
}
```

Covered in "Check for table existence in console before running plan containing CreateTable" as part of https://github.com/hyrise/hyrise/issues/1189 so it doesn't need a new issue.